libs/pageserver_api/src/key.rs

```diff
@@ -706,15 +706,49 @@ pub fn repl_origin_key_range() -> Range<Key> {
 /// Non inherited range for vectored get.
 pub const NON_INHERITED_RANGE: Range<Key> = AUX_FILES_KEY..AUX_FILES_KEY.next();
 /// Sparse keyspace range for vectored get. Missing key error will be ignored for this range.
-pub const NON_INHERITED_SPARSE_RANGE: Range<Key> = Key::metadata_key_range();
+pub const SPARSE_RANGE: Range<Key> = Key::metadata_key_range();
 
 impl Key {
     // AUX_FILES currently stores only data for logical replication (slots etc), and
     // we don't preserve these on a branch because safekeepers can't follow timeline
     // switch (and generally it likely should be optional), so ignore these.
     #[inline(always)]
     pub fn is_inherited_key(self) -> bool {
-        !NON_INHERITED_RANGE.contains(&self) && !NON_INHERITED_SPARSE_RANGE.contains(&self)
+        if self.is_sparse() {
+            self.is_inherited_sparse_key()
```

Comment:
nit: `is_sparse` is being checked twice

Reply:
Yes this is done intentionally because it's a public function. I can split out a private function do that we don't need to do the same work twice.

Reply:
fixed in https://github.com/neondatabase/neon/pull/10313/commits/2e69e9fbd81a7453dc4ede7ddfac9eb8f257859d